### PR TITLE
Fixing model nil cases

### PR DESCRIPTION
**Pull request by dphilla**

### CODE REVIEW

File: app/models/condition.rb [hunk 1]
```diff
@@ -49,10 +49,12 @@ def self.avg_rides_by_precip(start_of_range)
...
-      avg = total_trips.count / dates_in_temp_range.count
-
-      avg.round(2)
+      if dates_in_temp_range.count == 0
+        "no data"
+      else
+        avg = total_trips.count / dates_in_temp_range.count
+        avg.round(2)
+      end
     end
```
Comment (mimizotti) [inline app/models/condition.rb:58]: This block looks great.

File: app/models/condition.rb [hunk 2]
```diff
@@ -82,9 +84,12 @@ def self.avg_rides_by_windspeed(start_of_range)
...
```
Comment (mimizotti) [inline app/models/condition.rb:93]: I like this approach over the one we spoke about in person. Nice work.

File: app/models/condition.rb [hunk 3]
```diff
@@ -147,6 +155,10 @@ def self.weather_on_day_with_highest_rides
 
     def self.weather_on_day_with_lowest_rides
       date = Trip.group(:start_date).order('count_id DESC').count(:id).keys.last
-      Condition.find_by(date: date)
+      if Condition.find_by(date: date) == []
+        "no data"
+      else
+        Condition.find_by(date: date)
+      end
     end
```
Comment (mimizotti) [inline app/models/condition.rb:163]: Looks good.

**Review by mimizotti — APPROVED** [review]

File: app/models/condition.rb (hunk 1 above)
Comment (snayrouz) [reply app/models/condition.rb:58]: Looks great Dan. Loving the logic break down

**Pull request merged**